test/common/router/config_impl_test.cc

```diff
@@ -1490,7 +1525,7 @@ TEST_F(RouteMatcherTest, TestRouteList) {
   {
     Http::TestRequestHeaderMapImpl headers =
         genHeaders("lyft.com", "/new_endpoint/foo/match_header", "GET");
-    headers.setByKey("x-match-header", "matched");
+    headers.addCopy("x-match-header", "matched");
```

Comment:
Is this change necessary?

Reply:
Not necessary. When I set the header in TestMatchTreeDynamicCluster, I found `setByKey` may cause the lifetime problem and I noticed `setByKey` is barely used in Envoy other tests. So I also changed my previous code.

Reply:
yeah, `setByKey` is api from `TraceContext`. If you want keep this change, then `setCopy` would be better here to keep the semantic.

Reply:
Got it. Done.